backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/material/interceptor/transactionevent/HUDescriptorsFromHUAssignmentService.java

```diff
@@ -148,6 +148,7 @@ private Map<MaterialDescriptor, Collection<HUDescriptor>> createMaterialDescript
 
 			final BigDecimal quantity = huDescriptorsForCurrentProduct
 					.stream()
+					.filter(huDescriptor -> !huDescriptor.isExternalProperty())
```

Comment:
stupid question: why?

Reply:
Because the event generated here will be consumed in `de.metas.material.cockpit.stock.StockDataUpdateRequestHandler#handleDataUpdateRequest`, which will modify `MD_Stock.QtyOnHand`. This is the primary motivation of the issue, to prevent service returns count against qty on hand.

Reply:
@adi-stefan  then wondering why not renaming the `quantity` to `qtyOnHand` to better reflect that, wdyt?

Reply:
done.